apps/fas-jupyter-cs109b/Dockerfile

```diff
@@ -1,52 +1,86 @@
+# for libraries already installed via `mamba` 
 
 USER root
 
 RUN apt-get -y update
-RUN apt-get -y install libgl1-mesa-glx
-
-# Update conda
-RUN conda update --quiet --yes conda
-
-# Add extra conda channels
-RUN conda config --append channels conda-forge \
-    --append channels anaconda-fusion \
-    --append channels jmcmurray
-
-RUN conda install --quiet --yes \
-    'mkl-service' \
-    'gensim' \
-    && \
-    conda clean --al -f -y
-
-COPY requirements.txt . 
-RUN pip install --upgrade pip && \
-    pip install --no-cache-dir -r requirements.txt
-    
-RUN rm requirements.txt
-
-# Activate ipywidgets extension in the environment that runs the notebook server, and do some clean-up
-RUN jupyter nbextension enable --py widgetsnbextension --sys-prefix && \
-    npm cache clean --force && \
-    rm -rf "/home/${NB_USER}/.cache/yarn" && \
-    rm -rf "/home/${NB_USER}/.node-gyp" && \
-    fix-permissions "${CONDA_DIR}" && \
-    fix-permissions "/home/${NB_USER}"
-
-# Install facets which does not have a pip or conda package at the moment
-WORKDIR /tmp
-RUN git clone https://github.com/PAIR-code/facets.git && \
-    jupyter nbextension install facets/facets-dist/ --sys-prefix && \
-    rm -rf /tmp/facets && \
-    fix-permissions "${CONDA_DIR}" && \
-    fix-permissions "/home/${NB_USER}"
-
-# Import matplotlib the first time to build the font cache.
-ENV XDG_CACHE_HOME="/home/${NB_USER}/.cache/"
-
-RUN MPLBACKEND=Agg python -c "import matplotlib.pyplot" && \
-    fix-permissions "/home/${NB_USER}"
+RUN apt-get -y install \
+    gconf-service \
+    libgl1-mesa-glx \
+    libasound2 \
+    libatk1.0-0 \
+    libatk-bridge2.0-0 \
+    libc6 \
+    libcairo2 \
+    libcups2 \
+    libdbus-1-3 \
+    libexpat1 \
+    libfontconfig1 \
+    libgcc1 \
+    libgconf-2-4 \
+    libgdk-pixbuf-2.0-0 \
+    libglib2.0-0 \
+    libgtk-3-0 \
+    libnspr4 \
+    libpango-1.0-0 \
+    libpangocairo-1.0-0 \
+    libstdc++6 \
+    libx11-6 \
+    libx11-xcb1 \
+    libxcb1 \
+    libxcomposite1 \
+    libxcursor1 \
+    libxdamage1 \
+    libxext6 \
+    libxfixes3 \
+    libxi6 \
+    libxrandr2 \
+    libxrender1 \
+    libxss1 \
+    libxtst6 \
+    ca-certificates \
+    fonts-liberation \
+    libappindicator1 \
+    libnss3 \
+    lsb-release \
+    xdg-utils \
+    wget \
+    libcairo-gobject2 \
+    libxinerama1 \
+    libgtk2.0-0 \
+    libpangoft2-1.0-0 \
+    libthai0 \
+    libpixman-1-0 \
+    libxcb-render0 \
+    libharfbuzz0b \
+    libdatrie1 \
+    libgraphite2-3 \
+    libgbm1
 
 USER $NB_UID
 
 WORKDIR $HOME
+
+# Install Python 3 packages using mamba, following convention from origin image
+# RUN mamba install --quiet --yes \
+#     'package-name'
+
+# Install packages that can only be installed via pip
+# COPY requirements.txt .
+# RUN pip install --upgrade pip && \
+#     pip install --no-cache-dir -r requirements.txt
+
+# RUN rm requirements.txt
+
+COPY cs109b.yml .
+RUN conda env update --name base --file cs109b.yml
```

Comment:
Nice work getting this to install cleanly. Had such a tough time with this last year, it's a relief to see it simplified!